dask/dataframe/io/csv.py

```diff
@@ -26,15 +27,17 @@
 from ...delayed import delayed
 from ...utils import asciitable, parse_bytes
 from ..utils import clear_known_categories
-from ...blockwise import BlockwiseIO
+from ...blockwise import Blockwise, blockwise_token
 
 import fsspec.implementations.local
 from fsspec.compression import compr
 
 
-class CSVSubgraph(Mapping):
+class CSVFunctionWrapper:
```

Comment:
@jakirkham - This is the wrapper function I mentioned offline.  Note that the`__call__` definition needs to be smart enough to map the `key` to the data (even if trhe key was stringified by the scheduler).

Reply:
Got it. I think it's probably not difficult to have `Dispatch` support `str`s along with types. That would help us move away from custom objects a bit if that's causing us issues.

Reply:
We could also just have a `dict` of functions [like we do with serialization]( https://github.com/dask/distributed/blob/e9cd97f8fef88268f26d855b1f15c955fe7b37b5/distributed/protocol/serialize.py#L157-L167 )

Reply:
The client may need to persist various objects in these `FunctionWrapper` functions to enable to worker to translate a key to an output chunk/partition.  It feels like a unified dict of functions or Dispatch approach may get messy, but  I agree it is worth exploring something like that to provide a clear template for future Blockwise-IO implementations.